### PR TITLE
Feat/wc product brand

### PR DESCRIPTION
**Pull request by kzamanbd**
### All Submissions:

* [x] My code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)
* [x] My code satisfies feature requirements
* [x] My code is tested
* [ ] My code passes the PHPCS tests
* [ ] My code has proper inline documentation
* [ ] I've included related pull request(s) (optional)
* [x] I've included developer documentation (optional)
* [x] I've added proper labels to this pull request

### Related Pull Request(s)

* Full PR Link https://github.com/getdokan/dokan-pro/pull/4297

### Closes 

* Closes https://github.com/getdokan/dokan-pro/issues/4206

### How to test the changes in this Pull Request:

* Steps or issue link

### Changelog entry

1. Added functionality for product brands, including a new brand selection, brand edit and brand wise product filter option on the vendor dashboard.

### Before Changes

Describe the issue before changes with screenshots(s).

### After Changes

Describe the issue after changes with screenshot(s).

### Feature Video (optional)

Link of detailed video if this PR is for a feature.

### PR Self Review Checklist:

* Code is not following code style guidelines
* Bad naming: make sure you would understand your code if you read it a few months from now.
* KISS: Keep it simple, Sweetie (not stupid!).
* DRY: Don't Repeat Yourself.
* Code that is not readable: too many nested 'if's are a bad sign.
* Performance issues
* Complicated constructions that need refactoring or comments: code should almost always be self-explanatory.
* Grammar errors.

### FOR PR REVIEWER ONLY:

> As a reviewer, your feedback should be focused on the idea, not the person. Seek to understand, be respectful, and focus on constructive dialog.

> As a contributor, your responsibility is to learn from suggestions and iterate your pull request should it be needed based on feedback. Seek to collaborate and produce the best possible contribution to the greater whole.

* [ ] Correct — Does the change do what it’s supposed to? ie: code 100% fulfilling the requirements?
* [ ] Secure — Would a nefarious party find some way to exploit this change? ie: everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities?
* [ ] Readable — Will your future self be able to understand this change months down the road?
* [ ] Elegant — Does the change fit aesthetically within the overall style and architecture?

## Summary by CodeRabbit

- **New Features**
  - Introduced a dynamic, search-enabled dropdown for selecting multiple product brands in the product editor.
  - Added brand-based filtering options in product listings for easier product discovery.
  - Enabled product brand assignment during product updates to streamline brand management.

### CODE REVIEW

File: includes/Ajax.php [hunk 1]
```diff
@@ -714,6 +715,81 @@ public function dokan_json_search_products_tags() {
...
+        if ( $brands ) {
+            foreach ( $brands as $brand ) {
+                $brands_data[] = [ $brand->term_id, $brand->name ];
+                // Recursively get children
+                $this->get_child_terms_recursive( $brand->term_id, $brands_data, 1 );
+            }
+        }
+
+        echo wp_json_encode( $brands_data );
+        die;
```
Comment (mrabbani) [inline includes/Ajax.php:790]: Pls use `wp_send_json( $brands_data ) `

**Review by mrabbani — CHANGES_REQUESTED** [review]
